specs/Qowaiv.Validation.Specs/Abstractions/Result_specs.cs

```diff
@@ -99,14 +100,37 @@ public class Casting
     public void Implicit_from_T_to_Result_of_T_is_supported()
     {
         Result<int> result = 17;
-        Assert.That(result.Value, Is.EqualTo(17));
+        result.Should().BeValid().Value.Should().Be(17);
     }
 
     [Test]
     public void Explicit_from_Result_of_T_to_T_is_supported()
     {
         var result = Result.For(666);
-        Assert.That((int)result, Is.EqualTo(666));
+        result.Should().BeValid().Value.Should().Be(666);
+    }
+
+    [Test]
+    public void With_method_to_Result_of_valid__wiht_subtype_of_T_is_supported()
+    {
+        var value = XDocument.Parse("<root />");
+        Result.For(value).Cast<XNode>().Should().BeValid()
+            .Value.Should().BeSameAs(value);
+    }
+
+
+    [Test]
+    public void With_method_to_Result_of_invalid_with_subtype_of_T_is_supported()
+    {
+        Result.WithMessages<XDocument>(Error1).Cast<XNode>().Should().BeInvalid();
+    }
+
+    [Test]
+    public void With_method_is_not_supported_TOut_not_bein_subtype_of_TModel()
```

Comment:
bein -> being
Also the is_supported is on the end on the previous tests. Also is_not_supported here at the end for consistency?